### ISSUE TITLE
Android - RN .73.6 - Incompatibility between JVM settings

### ISSUE THREAD

**Issue by gladiuscode** [edeckers/react-native-blob-courier#257]
**Describe the bug**
I've tried to add this library in my react-native application (version .73.6) but Android can't compile due to a conflict between JVM settings required by Kotlin.
Java 17 is now a requirement due to [React Native migration](https://reactnative.dev/blog/2023/12/06/0.73-debugging-improvements-stable-symlinks#android-14-support) to Kotlin Template and Android 14 support.
This project sets the following block options in its build.gradle that make it incompatible with react-native itself:

```
compileOptions {
    sourceCompatibility JavaVersion.VERSION_11
    targetCompatibility JavaVersion.VERSION_11
}
kotlinOptions {
    jvmTarget = JavaVersion.VERSION_11.toString()
}
```

I've looked into other projects leveraging Kotlin for guidance on how to possibile fix it and I've found that [react-native-webview](https://github.com/react-native-webview/react-native-webview#readme) doesn't set such blocks even tho it leverages Kotlin.

**To Reproduce**
1. Init a new react-native project and add this library
2. Start compilation
4. See error below:
```
Execution failed for task ':react-native-blob-courier:compileDebugKotlin'.
> 'compileDebugJavaWithJavac' task (current target is 17) and 'compileDebugKotlin' task (current target is 11) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain
```

**Expected behavior**
It should compile successfully with react-native .73.6.

**Smartphone (please complete the following information):**
 - Device: Pixel 7 Emulator
 - OS: Android
 - Version 14 (API 34)

**Comment by gladiuscode** [comment]
I've leveraged [patch-package](https://github.com/ds300/patch-package) to apply the following patch:

```
diff --git a/node_modules/react-native-blob-courier/android/build.gradle b/node_modules/react-native-blob-courier/android/build.gradle
index 255fb12..9d0c4e0 100644
--- a/node_modules/react-native-blob-courier/android/build.gradle
+++ b/node_modules/react-native-blob-courier/android/build.gradle
@@ -75,13 +75,6 @@ android {
   lintOptions {
     disable 'GradleCompatible'
   }
-  compileOptions {
-    sourceCompatibility JavaVersion.VERSION_11
-    targetCompatibility JavaVersion.VERSION_11
-  }
-  kotlinOptions {
-    jvmTarget = JavaVersion.VERSION_11.toString()
-  }
 }

 repositories {
```

It removes compileOptions and kotlinOptions from build.gradle.
I haven't got a real experience with android/kotlin so it would be nice if someone can confirm that this does indeed fix the issue without breaking something else.

I've set this patch and I can now run my project successfully.

I'll open a PR this evening :)

**Comment by edeckers** [comment]
Hey @gladiuscode, thank you for reporting the issue! These changes seem like something I should be able to merge without any problems. If you create a PR and the testset succeeds, I'll accept it :)

**Comment by gladiuscode** [comment]
> Hey @gladiuscode, thank you for reporting the issue! These changes seem like something I should be able to merge without any problems. If you create a PR and the testset succeeds, I'll accept it :)

Hi @edeckers!
I've opened it, thank you!

**Comment by edeckers** [comment]
Thank you @gladiuscode! Your changes were merged and released in version 5.0.1 :)
